src/components/layout/GradientScrollView.tsx

```diff
@@ -1,166 +1,302 @@
 import * as React from "react";
-import { useMemo } from "react";
+import { ComponentProps, PropsWithChildren, useMemo, useState } from "react";
+import {
+  ColorValue,
+  LayoutChangeEvent,
+  LayoutRectangle,
+  StyleSheet,
+  View
+} from "react-native";
+import { easeGradient } from "react-native-easing-gradient";
+import LinearGradient from "react-native-linear-gradient";
 import Animated, {
   Easing,
+  Extrapolate,
+  interpolate,
   useAnimatedScrollHandler,
   useAnimatedStyle,
-  useSharedValue,
-  withTiming
+  useSharedValue
 } from "react-native-reanimated";
 import { useSafeAreaInsets } from "react-native-safe-area-context";
 import {
+  IOColors,
   IOSpacer,
   IOSpacingScale,
   IOVisualCostants,
-  buttonSolidHeight
+  hexToRgba,
+  useIOTheme
 } from "../../core";
 import { WithTestID } from "../../utils/types";
-import GradientBottomActions from "./GradientBottomActions";
-
-export type GradientScrollView = WithTestID<{
-  children: React.ReactNode;
-  excludeSafeAreaMargins?: boolean;
-  debugMode?: boolean;
-  // Accepted components: ButtonSolid, ButtonLink
-  // Don't use any components other than this, please.
-  primaryActionProps: GradientBottomActions["primaryActionProps"];
-  secondaryActionProps?: GradientBottomActions["secondaryActionProps"];
-}>;
-
-// Extended gradient area above the actions
-export const gradientSafeArea: IOSpacingScale = 96;
-// End content margin before the actions
+import { ButtonLink, ButtonOutline, ButtonSolid } from "../buttons";
+import { VSpacer } from "../spacer";
+
+type GradientScrollActions =
+  | {
+      type: "SingleButton";
+      primary: Omit<ComponentProps<typeof ButtonSolid>, "fullWidth">;
+      secondary?: never;
+      tertiary?: never;
+    }
+  | {
+      type: "TwoButtons";
+      primary: Omit<ComponentProps<typeof ButtonSolid>, "fullWidth">;
+      secondary: ComponentProps<typeof ButtonLink>;
+      tertiary?: never;
+    }
+  | {
+      type: "ThreeButtons";
+      primary: Omit<ComponentProps<typeof ButtonSolid>, "fullWidth">;
+      secondary: Omit<ComponentProps<typeof ButtonOutline>, "fullWidth">;
+      tertiary: ComponentProps<typeof ButtonLink>;
+    };
+
+type GradientScrollView = WithTestID<
+  PropsWithChildren<{
+    actionsProps: GradientScrollActions;
```

Comment:
What about making this not mandatory in order to be able to use the ScrollView as a generic one without the actions?

```suggestion
    actionsProps?: GradientScrollActions;
```

Reply:
@CrisTofani Addressed in the following commit → https://github.com/pagopa/io-app-design-system/pull/243/commits/8d550e438ff6cc542bedba94fa365bafda2915a7 (the Typescript error popped up again, so casting was necessary)